File: Demo/Sources/ViewControllers/DataView/Simple/DataProvider/SimpleDataProvider.m
//
//  Globus
//

#import "SimpleDataProvider.h"

@implementation SimpleDataProvider

+ (instancetype)dataProvider {
    SimpleDataProvider* dataProvider = [self new];
    dataProvider.canReload = YES;
    return dataProvider;
}

- (id)modelWithJsonObject:(id)jsonObject {
    return [SimpleDataProviderModel modelWithJson:jsonObject];
}

@end

@implementation SimpleDataProviderModel

#pragma mark - GLBModel

+ (NSDictionary< NSString*, GLBModelJson* >*)jsonMap {
    return @{
        @"uid" : [GLBModelJsonString jsonWithPath:@"id"],
        @"url" : [GLBModelJsonUrl jsonWithPath:@"avatar"],
        @"firstName" : [GLBModelJsonString jsonWithPath:@"first_name"],
        @"lastName" : [GLBModelJsonString jsonWithPath:@"last_name"]
    };
}

@end

File: Demo/Sources/ViewControllers/DataView/Simple/DataProvider/SimpleDataProvider.swift
//
//  Globus
//

import Globus
import GlobusSwifty

class SimpleDataProvider: GLBLocalSimpleDataProvider {
    
    public override init() {
        super.init(baseFileName: "SimpleDataProvider")
        self.canReload = true
    }
    
    public override init(baseFileName: String) {
        super.init(baseFileName: baseFileName)
        self.canReload = true
    }
    
    public override func model(withJsonObject jsonObject: Any) -> Any? {
        return SimpleDataProviderModel.model(jsonObject: jsonObject)
    }
    
}

class SimpleDataProviderModel : GLBSwiftyModel {
    
    public var uid: String = ""
    public var url: URL?
    public var firstName: String = ""
    public var lastName: String = ""
    
    override func fromJson(json: GLBJson) {
        self.uid <<< (json, "id")
        self.url <<< (json, "avatar")
        self.firstName <<< (json, "first_name")
        self.lastName <<< (json, "last_name")
    }
    
}
